middleware/cors.go

```diff
@@ -102,40 +109,50 @@ func CORSWithConfig(config CORSConfig) echo.MiddlewareFunc {
 			origin := req.Header.Get(echo.HeaderOrigin)
 			allowOrigin := ""
 
-			// Check allowed origins
-			for _, o := range config.AllowOrigins {
-				if o == "*" && config.AllowCredentials {
-					allowOrigin = origin
-					break
-				}
-				if o == "*" || o == origin {
-					allowOrigin = o
-					break
-				}
-				if matchSubdomain(origin, o) {
-					allowOrigin = origin
-					break
-				}
-			}
-
-			// Check allowed origin patterns
-			for _, re := range allowOriginPatterns {
-				if allowOrigin == "" {
-					didx := strings.Index(origin, "://")
-					if didx == -1 {
-						continue
+			if config.AllowOriginFunc == nil {
```

Comment:
I'd prefer using `config.AllowOriginFunc != nil` and having the new custom code check in front of the existing code, as it actually overrides the current checks if it is defined.

Like:
```go
if config.AllowOriginFunc != nil {
  // use the custom function
} else {
  // use existing code
}
```